ros2bag_tools/ros2bag_tools/filter/drop.py

```diff
@@ -20,30 +21,49 @@ class DropFilter(FilterExtension):
     """Drop X out every Y message of a topic."""
 
     def __init__(self):
-        self._topic = None
+        self._topics: Sequence[str] = []
         self._x = 0
         self._y = 0
         self._i = 0
 
     def add_arguments(self, parser):
-        parser.add_argument('-t', '--topic', help='topic to drop messages from')
         parser.add_argument(
-            '-x',
-            type=int,
-            required=True,
-            help='count of messages out of ytodrop'
+            "-t",
+            "--topics",
+            nargs="+",
+            help=(
+                "Topics to drop messages from. Use 'all' to drop messages from all "
+                "topics in the rosbag"
+            ),
+        )
+        parser.add_argument(
+            "-x", type=int, required=True, help="count of messages out of ytodrop"
+        )
+        parser.add_argument(
+            "-y", type=int, required=True, help="module of the message counter"
         )
-        parser.add_argument('-y', type=int, required=True, help='module of the message counter')
 
     def set_args(self, _metadata, args):
-        self._topic = args.topic
+        self._topics = args.topics
         self._x = args.x
         self._y = args.y
         assert self._x <= self._y
 
+    def _get_topic_fn(self):
```

Comment:
The alternative would be to format it as a lengthy if statement like

```python
if (len(self._topics) == 1 and self._topics[0] == "all") or topic in self._topics:
    ...
```

I just found it cleaner to define a named function and encapsulating the logic there. This way the user knows from the method's name/docstring what this condition is about

Reply:
I rather meant why not check it in the function instead of returning a function that provides the check?

```python
def is_drop_topic(self, topic):
  if len(self._topics) == 1 and self._topics[0] == "all":
    return True
  return topic in self._topics

if self.is_drop_topic(topic):
  ...

# besides, that would be one possibility, but fine with either
return self._topics == ['all'] or topic in self._topics
 ....
```